### PR TITLE
first stab at sql based seed saving

### PR DESCRIPTION
**Pull request by jbarnicle**
@LinuxBozo @rogeruiz - I need to double check the bean mappings but this should be pretty close.

### CODE REVIEW

**Comment by jbarnicle** [comment]
So I was looking at the existing flow, and I think if we 

- created a flow element called something like `CheckForMFA` that simply did a query to check if there is an existing secret 
- have the password flow forward to that flow element instead of the `DisplayTotpFom` flow element
- had `CheckForMFA` forward to `DisplayTotpForm` or `RegisterToken` depending on result

we'd be really close to having the finalized flow.

File: totpauth-impl/src/main/resources/conf/authn/totp-authn-config.xml [hunk 1]
```diff
@@ -19,6 +23,11 @@
 	<bean id="shibboleth.authn.mongoDbName" class="java.lang.String" c:_0="tokenSeed" />
 	<bean id="shibboleth.authn.mongoDbUrl" class="java.lang.String" c:_0="127.0.0.1" />
 
+
+	<!-- MongoDB related fields -->
+	<bean id="shibboleth.authn.mongoDbName" class="java.lang.String" c:_0="tokenSeed" />
+	<bean id="shibboleth.authn.mongoDbUrl" class="java.lang.String" c:_0="127.0.0.1" />
```
Comment (rogeruiz) [inline totpauth-impl/src/main/resources/conf/authn/totp-authn-config.xml]: Not sure why these are in here twice @jbarnicle ?

File: totpauth-impl/src/main/resources/conf/authn/totp-authn-config.xml [hunk 2]
```diff
@@ -11,6 +11,10 @@
 	<!-- Names of form fields to pull tokenNumber -->
 	<bean id="shibboleth.authn.tokenCodeField" class="java.lang.String" c:_0="j_tokenNumber" />
 
+	<!-- TOTP Seed DB related fields -->
+	<bean id="shibboleth.authn.seedDBTableName" class="java.lang.String" c:_0="dbtablename"/>
+	<bean id="shibboleth.authn.usernameColumnName" class="java.lang.String" c:_0="email"/>
+	<bean id="shibboleth.authn.seedColumnName" class="java.lang.String" c:_0="totpseed"/>
```
Comment (rogeruiz) [inline totpauth-impl/src/main/resources/conf/authn/totp-authn-config.xml]: These database columns will need to match what's in 18F/cg-provision#118.

https://github.com/LinuxBozo/cg-provision/blob/d7b6a8989209774f0b3167815b5d8555b7eb59a6/ci/scripts/create-and-update-db.sh#L26

File: totpauth-impl/src/main/resources/conf/authn/totp-authn-beans.xml [hunk 3]
```diff
@@ -21,14 +21,15 @@
...
         p:gAuth-ref="shibboleth.totp.gAuth"
-        p:ldapTemplate-ref="ldapTemplate"
-        p:tokenCodeField-ref="shibboleth.authn.tokenCodeField">
-        <constructor-arg ref="shibboleth.authn.seedAttribute" />
-        <constructor-arg ref="shibboleth.authn.userAttribute" />
+        p:seedDataSource-ref="seedDataSource"
+        p:tokenCodeField-ref="shibboleth.authn.tokenCodeField"
+        p:seedDBTableName-ref="shibboleth.authn.seedDBTableName"
+        p:usernameColumnName-ref="shibboleth.authn.usernameColumnName"
+        p:seedColumnName-ref="shibboleth.authn.seedColumnName">
```
Comment (rogeruiz) [inline totpauth-impl/src/main/resources/conf/authn/totp-authn-beans.xml]: Doesn't seem like the `RegisterNewSeedSql` class supports `backup_code` just yet, but shouldn't be a big deal for now.

File: totpauth-impl/src/main/resources/conf/authn/totp-authn-config.xml (hunk 2 above)
Comment (LinuxBozo) [inline totpauth-impl/src/main/resources/conf/authn/totp-authn-config.xml]: They don't have to exactly.. can override in cg-deploy repo..

File: totpauth-impl/src/main/resources/conf/authn/totp-authn-beans.xml (hunk 3 above)
Comment (LinuxBozo) [inline totpauth-impl/src/main/resources/conf/authn/totp-authn-beans.xml]: Right.. created the table knowing future plans.. no need to support backup codes at all right this second..

Comment (jbarnicle) [inline totpauth-impl/src/main/resources/conf/authn/totp-authn-beans.xml]: Yup. Figured this was MVP

**Comment by rogeruiz** [comment]
Seems like since we're going to configure this in `18F/shibboleth-boshrelease` none of the updates I was proposing in here are necessary. Will update it in the `boshrelease`. 👍

**Pull request merged**